base/libgit2/types.jl

```diff
@@ -174,7 +174,50 @@ function RemoteCallbacks(credentials::Ptr{Void}, payload::Ref{Nullable{AbstractC
     RemoteCallbacks(credentials=credentials_cb(), payload=pointer_from_objref(payload))
 end
 
-if LibGit2.version() >= v"0.24.0"
+immutable ProxyOptions
```

Comment:
release-0.5 shouldn't match what's on master, it should remain close to 0.5.0. just change msys_build.sh to download 0.5-latest and you won't need to touch libgit2 at all